### PR TITLE
gpt-auto: expand the loader partition UUID check to include XBOOTLDR

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/gpt-auto-generator/gpt-auto-generator.c
Comment (yuwata) [inline src/gpt-auto-generator/gpt-auto-generator.c]: The error will be ignored now. So, please downgrade the log level. I think, `log_debug_errno()` is sufficient. At most, `log_info_errno()`.
Then, append ", ignoring" in the message.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Review by yuwata — APPROVED** [review]
LGTM. But I'd like to wait for @poettering's final review.

**Comment by YHNdnzj** [comment]
Just rebased to resolve merge conflicts. No functional change :)

File: src/gpt-auto-generator/gpt-auto-generator.c
Comment (poettering) [inline src/gpt-auto-generator/gpt-auto-generator.c]: indentation to 4 space is a bit weird, usually we'd indent this deeper (16spaces?), but doesn't matter

File: src/gpt-auto-generator/gpt-auto-generator.c
Comment (poettering) [inline src/gpt-auto-generator/gpt-auto-generator.c]: please still use log_debug_errno() here, and pass `r` here, even if you don't use `%m` in the format string. the error number is still included in the structure log message as a separate field then, which is a good thing.

i.e. log_xyz_errno() should be used primarily when there's an error code we are acting on, and only secondarily to make %m output something nice

File: src/gpt-auto-generator/gpt-auto-generator.c
Comment (poettering) [inline src/gpt-auto-generator/gpt-auto-generator.c:765]: could you add a comment somewhere here:

```
/* Let's check if this is actually the ESP that was used for booting. If so, great. If not, let's see if the XBOOTLDR partition we found was used for booting. If so, let's use the ESP from the same disk too. If not, don't mount it */

File: src/gpt-auto-generator/gpt-auto-generator.c
Comment (poettering) [inline src/gpt-auto-generator/gpt-auto-generator.c]: this really should check `m->partitions[PARTITION_XBOOTLDR].found` before using `m->partitions[PARTITION_XBOOTLDR].uuid`

**Review by poettering — CHANGES_REQUESTED** [review]
so this looks great, some comments though.

**Comment by YHNdnzj** [comment]
Thank you for the review :)
I’m currently on vacation though. Will update when I get back.

File: src/gpt-auto-generator/gpt-auto-generator.c
Comment (poettering) [inline src/gpt-auto-generator/gpt-auto-generator.c]: doesn't matter, but we usually prefer shallow indentation, hence would usually write this as :

```c
if (r == -ENOENT) {
        …
}
if (r < 0) {
        …
}
`` 

but doesn't matter at all

File: src/gpt-auto-generator/gpt-auto-generator.c
Comment (poettering) [inline src/gpt-auto-generator/gpt-auto-generator.c]: "anyway" suggests this was a problem of some sort. I think the message generally makes sense for informational reasons, but I'd drop the ", proceeding anyway".

**Review by poettering — APPROVED** [review]
looks good, just some minor things that really don't matter

**Pull request merged**
